### ISSUE TITLE
Simplify conversion of meta task to backlog tasks

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#1031]
**Is your feature request related to a problem? Please describe.**
- It seems to take a lot of effort to fully convert a meta task into the format that's fit for a settlement task.

**Describe the solution you desire to see**
- Use a list of standard params for all settlement capable task.
- Capture the entity of a settlement task.

**Specifications  (please complete the following information):**
 - pre-3.7.0 build 8627

**Additional context**
- What criteria are there for converting a normal meta task to a settlement task ?
- Once the conversion is done, is there a way for a settler to select that meta task on his own ?
- Is there a quicker way to convert a meta task, rather than manually converting each ?
- Is it possible to create a template that read a xml file for generating a brand new settlement task with a few basic predefined functions ?

**Comment by bevans2000** [comment]
Good idea to add this to the Ratings project

**Comment by mokun** [comment]
So, will the process by any chance be simplified further ?

**Comment by bevans2000** [comment]
> So, will the process by any chance be simplified further ?

I don't know, you reported it 😄 
Generally I'm working on consolidating the logic as there is massive inconsistency and duplication in the MetaTasks.

**Comment by bevans2000** [comment]
What's the goal of this issue? Creating SettlementMetaTask is not simple because it is a complex optimisation. I don't see how we can simplify it that much so let's close this.
Nevertheless, nearly all that can be converted have now been converted.

**Comment by mokun** [comment]
Q: is each SettlementMetaTask generated only once on each frame for all settlers ? 

For instance, in case of tending a greenhouse, in the past, each qualified settler would check on the same greenhouse for the need of tending. So in each frame, say, if you have a large settlement with 100 settlers. 50 of them might check on the same inflatable greenhouse and it's a waste of cpu.

I imagine SettlementMetaTask would only need to be created only once on each frame it would be good for the entire frame.

Also, another approach is that if the task is not taken in by anyone, it may simply be left being posted in the backlog with an expiration date or a timer countdown, it will save a lot of cpu cycles.

I wonder if you may be interested in working out the 2nd approach.

**Comment by bevans2000** [comment]
Yes you are  correct Settlement meta tasks only get evaluate once per frame and are shared by then 100 settlers. Each settler does have to score the shared task to their abilities but this is  very lightweight. The shared also have a demand count so the same shared Task can be claimed multiple times per frame without needing to recalculate. Greenhouse is a good example of this I think.
The shared list is not actually evualted every frame  but on demand when needed but the shared list of tasks remains valjd for one frame. So if there is a frame where every is on Task and does not need a new Task, the shared list is not created further saving CPU.
I think some stats at the time and only 60% of the frame actually needed the shared Task list.
Hence we could add an age of each shared Task but that would be extra complication and I was not sure if it was worth it. The saving of CPU I achieved with this simple approach was pretty good.
